cfgmgr/portmgr.cpp

```diff
@@ -76,6 +77,47 @@ bool PortMgr::setPortAdminStatus(const string &alias, const bool up)
     return true;
 }
 
+bool PortMgr::setPortDHCPMitigationRate(const string &alias, const string &dhcp_rate_limit)
+{
+    stringstream cmd;
+    string res, cmd_str;
+    int ret;
+    int byte_rate = stoi(dhcp_rate_limit) * 406;
+
+    if (dhcp_rate_limit != "0")
+    {
+        // tc qdisc add dev <port_name> handle ffff: ingress
+        // &&
+        // tc filter add dev <port_name> protocol ip parent ffff: prio 1 u32 match ip protocol 17 0xff match ip dport 67 0xffff police rate <byte_rate>bps burst <byte_rate>b conform-exceed drop
+        cmd << TC_CMD << " qdisc add dev " << shellquote(alias) << " handle ffff: ingress" << " && " \
+            << TC_CMD << " filter add dev " << shellquote(alias) << " protocol ip parent ffff: prio 1 u32 match ip protocol 17 0xff match ip dport 67 0xffff police rate " << to_string(byte_rate) << "bps burst " << to_string(byte_rate) << "b conform-exceed drop";
+        cmd_str = cmd.str();
+        ret = swss::exec(cmd_str, res);
+        SWSS_LOG_WARN("ret Value in setPortDHCPMitigationRate is  ret:%d,", ret);
+
+        if (!ret)
+        {   
+            SWSS_LOG_INFO("writing dhcp_rate_limit to appl_db");
+            return writeConfigToAppDb(alias, "dhcp_rate_limit", dhcp_rate_limit);
+        }
+        else if (!isPortStateOk(alias))
+        {
+            // Can happen when a DEL notification is sent by portmgrd immediately followed by a new SET notif
+            SWSS_LOG_WARN("Setting dhcp_rate_limit to alias:%s netdev failed with cmd:%s, rc:%d, error:%s", alias.c_str(), cmd_str.c_str(), ret, res.c_str());
+            return false;
+        }
+    }
+    else
+    {
+        // tc qdisc del dev <port_name> handle ffff: ingress
+        cmd << TC_CMD << " qdisc del dev " << shellquote(alias) << " handle ffff: ingress";
+        cmd_str = cmd.str();
+        ret = swss::exec(cmd_str, res);
```

Comment:
Are we not checking the value of ret here? What is the action if this command fails? Does it need to be logged?